server/src/main/java/org/opensearch/extensions/rest/AuthorizationRequest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions.rest;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.transport.TransportRequest;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Request to evaluate user privileges for extension actions
+ *
+ * @opensearch.internal
+ */
+public class AuthorizationRequest extends TransportRequest {
+
+    private String extensionUniqueId;
```

Comment:
Is there a stronger way we can identity a request from an extension?  @saratvemulapalli

Reply:
Removed references to extensions here. This now uses `PrincipalIdentifierToken`, if we generate different tokens for each extension then we may be able to get this implicitly.